packages/controllers/src/snaps/SnapController.ts

```diff
@@ -2071,6 +1984,32 @@ export class SnapController extends BaseController<
     return { manifest, sourceCode, svgIcon };
   }
 
+  /**
+   * Map initial permissions as defined in a Snap manifest to something that can
+   * be processed by the PermissionsController. Each caveat mapping function
+   * should return a valid permission caveat value.
+   *
+   * This function does not validate the caveat values, since that is done by
+   * the PermissionsController itself, upon requesting the permissions.
+   *
+   * @param initialPermissions - The initial permissions to process.
+   * @returns The processed permissions.
+   * @private
+   */
+  private processSnapPermissions(
+    initialPermissions: RequestedSnapPermissions,
+  ): RequestedSnapPermissions {
+    return Object.fromEntries(
```

Comment:
Wasn't there a problem with us using `fromEntries`? Or was that fixed?

Reply:
This seems to work, so I'm not sure?